bundler/lib/bundler/source/git/git_proxy.rb

```diff
@@ -181,6 +181,14 @@ def clone_needs_extra_fetch?
 
             if err.include?("Could not find remote branch")
               raise MissingGitRevisionError.new(command_with_no_credentials, nil, explicit_ref, credential_filtered_uri)
+            elsif err.include?("dumb http transport does not support shallow capabilities")
```

Comment:
If the problem is with `http` protocol, wouldn't be possible to detect this earlier in `depth_args` based on the URI provided and don't even try with `--depth` for http ones?

Reply:
We tried this. GitHub works with http, but the example in the issue fails. It has to do with how it's being hosted. This allows it to try first and then remove if it fails.

Reply:
So does this rely on host not on the git itself? If so, isn't there better way how to react to this (like exit code of the command) to prevent vendoring 3rd party error message (which could change at any time)?

Reply:
https://github.com/CocoaPods/cocoapods-downloader/pull/65/files looks similar, seems this is good approach for this :white_check_mark:

Reply:
I like their regex. Let's borrow it.

```
/^fatal:.*does not support (--depth|shallow capabilities)$/
```

Reply:
Maybe this can be completely moved into the `else` clause, so that we don't duplicate the `raise GitCommandError.new(command_with_no_credentials, path, err)` line.

Reply:
I think moving it to the else branch makes sense. That way it's also not dependent on the text of the error message. In the git source code, that error message is marked for translation, so it may not be consistent for all users. Also, even if some other error cause the code to go into the else branch, my understanding is that removing the depth argument wouldn't cause problems for the next retries. 

Added that change.

Reply:
Actually my suggestion was just to move the regexp check on the error message to inside the `else` branch:

```ruby
else
  if err.include?("dumb http transport does not support shallow capabilities")
    idx = command.index("--depth")
    if idx
      command.delete_at(idx)
      command.delete_at(idx)
      command_with_no_credentials = check_allowed(command)
      err += "Retrying without --depth argument."
    end
  end
  
  raise GitCommandError.new(command_with_no_credentials, path, err)
```

But maybe retrying every time fetching with `--depth` fails makes sense too!

Reply:
Ah, I see what you meant. I think with the translation issue it would make sense to remove the `--depth` argument for any failure that hits the `else` branch, but I can make that change if it's preferred.

Reply:
I'm happy to try the current way, the potential translation issue makes sense to me, I'll set this as resolved!